hc/api/management/commands/sendreports.py

```diff
@@ -28,27 +27,61 @@ def add_arguments(self, parser):
         )
 
     def handle_one_run(self):
+        """
+        Method to send emails for daily, weekly or monthly
+        based on user preference. Method is background worker.
+        """
+        # get current time
+
         now = timezone.now()
-        month_before = now - timedelta(days=30)
 
-        report_due = Q(next_report_date__lt=now)
-        report_not_scheduled = Q(next_report_date__isnull=True)
+        # get time stamp for day before to
+        # ensure user only gets email at the earliest at
+        # least 1 day after subscribing
+        number_of_days = {
+            "daily": 1,
+            "weekly": 7,
+            "monthly": 30
+        }
+
+        sent_list = []
 
-        q = Profile.objects.filter(report_due | report_not_scheduled)
-        q = q.filter(reports_allowed=True)
-        q = q.filter(user__date_joined__lt=month_before)
-        sent = 0
-        for profile in q:
-            if num_pinged_checks(profile) > 0:
-                self.stdout.write(self.tmpl % profile.user.email)
-                profile.send_report()
-                sent += 1
+        for item in number_of_days.keys():
 
-        return sent
+            period_before = now - timedelta(minutes=number_of_days[item])
+            # initialise db query conditions for next report date
+            report_due = Q(next_report_date__lt=now)
```

Comment:
I belive this sequence of reports can be joined in a query. saves system cycles and memories